### PR TITLE
Fix mypy issues and re-activate

### PR DESCRIPTION
**Pull request by mikebauer**
## Description of change
Fix all the mypy errors we've been encountering and re-activate the check.

## Guide to reproduce test results.
```bash
trunk check --all
bazel test //...
```
## Checklist:
- [x] I have self-reviewed this change.
- [x] I have tested this change.
- [x] This change is covered by tests that are already landed, or in this PR.

### CODE REVIEW

File: .github/workflows/lint-and-codecov.yml [hunk 1]
```diff
@@ -24,21 +24,13 @@ jobs:
...
-        run: bazel coverage ... --combined_report=lcov --java_runtime_version=remotejdk_11 --nocache_test_results --test_tag_filters="-integration"
-
-      - uses: codecov/codecov-action@v3
-        with:
-          token: ${{ secrets.CODECOV_TOKEN }}
-
       - name: Trunk Check
         uses: trunk-io/trunk-action@v1
+        with:
+          check-mode: all
```
Comment (mikebauer) [inline .github/workflows/lint-and-codecov.yml:36]: TODO(michael) Revert changes to this file

File: resim/metrics/python/unpack_metrics.py [hunk 2]
```diff
@@ -395,10 +397,14 @@ def _unpack_plotly_metric(
...
     unpacked: ImageMetric,
-    id_to_unpacked_metrics_data: dict[uuid.UUID, MetricsData],
+    id_to_unpacked_metrics_data: dict[uuid.UUID, BaseMetricsData],
 ) -> None:
     image_data = metric.metric_values.image_metric_values
     data = id_to_unpacked_metrics_data[_unpack_uuid(image_data.image_data_id.id)]
+    if not isinstance(data, ExternalFileMetricsData):
+        raise ValueError(
+            "MetricsData for image metric must be ExternalFileMetricsData!"
+        )
```
Comment (mikebauer) [inline resim/metrics/python/unpack_metrics.py:407]: May have to add a test for this case to get coverage.

Comment (mikebauer) [reply resim/metrics/python/unpack_metrics.py:407]: Yep. Fix incoming.

Comment (mikebauer) [reply resim/metrics/python/unpack_metrics.py]: Went with a cast for now instead.

File: .github/workflows/lint-and-codecov.yml (hunk 1 above)
Comment (mikebauer) [reply .github/workflows/lint-and-codecov.yml]: Done

**Review by iainjwhiteside — APPROVED** [review]

**Pull request merged**